WordPress/src/main/java/org/wordpress/android/WordPress.java

```diff
@@ -372,6 +372,9 @@ private void endMigration() {
 
     public static void registerMigrationListener(MigrationListener listener) {
         sMigrationListener = listener;
+        if (sIsMigrationError) {
+            sMigrationListener.onError();
```

Comment:
>  I generally have this idea that in order for a method to accomplish its mission, first thing to do inside is to validate it has everything needed in order to go ahead.

I agree when you deliberately use null as a state (aka the "billion dollar mistake"), but in other cases we'll just hide bugs behind null checks. `@NonNull` annotation is better than nothing but it's just used by tools (AS and findbugs) so we can rely on it to prevent problems. I ❤️ Kotlin and its nullsafety for this.